experiments/TestCase/solid_body_rotation_mountain.jl

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/env julia --project
+using ClimateMachine
+ClimateMachine.init(parse_clargs = true)
+
+using ClimateMachine.Atmos
+using ClimateMachine.Orientations
+using ClimateMachine.ConfigTypes
+using ClimateMachine.NumericalFluxes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.SystemSolvers: ManyColumnLU
+using ClimateMachine.Mesh.Filters
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.Mesh.Interpolation
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.VariableTemplates
+using ClimateMachine.Thermodynamics: air_density, total_energy
+
+using LinearAlgebra
+using StaticArrays
+using Test
+
+using CLIMAParameters
+using CLIMAParameters.Planet: day, planet_radius
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+function setmax(f, r_inner, r_outer)
+    function setmaxima(a, b, c)
+        return f(
+            a,
+            b,
+            c,
+            max(abs(a), abs(b), abs(c));
+            r_inner = r_inner,
+            r_outer = r_outer,
+        )
+    end
+    return setmaxima
+end
+
+function cubedshelltopowarp(
+    a,
+    b,
+    c,
+    R = max(abs(a), abs(b), abs(c));
+    r_inner = _planet_radius,
+    r_outer = _planet_radius + domain_height,
+)
+
+    function f(sR, ξ, η, faceid)
```

Comment:
It seems the cubed sphere setup should be going somewhere where it can be reused more generally, not into this test setup.

Reply:
The following functionalities are moved to `Topologies.jl`:
- cubed sphere set up and topowarp function
- helper functions that computes lat/long
- function that generate the dcmip 2-0-0 mountain (since it is an archived dcmip case)

In the experiment driver, users can specify the topography (either choosing from what we have in `Topologies.jl` or define their own following the `AnalyticalTopography` type) meshwarp argument.